# This patch file is from Qualcomm Technologies, Inc. and is provided under the following license:
#
# Copyright (c) 2025 Qualcomm Technologies, Inc. All rights reserved.
#
# SPDX-License-Identifier: BSD-3-Clause-Clear

From 87ce5c6f95bb1e6277f8c978fc6b1a473334e2f8 Mon Sep 17 00:00:00 2001
From: Lei wang <leiwan@codeaurora.org>
Date: Sat, 25 May 2019 08:52:39 +0800
Subject: [PATCH 1/2] avb: bring up keymaster for LV

Always enable keymaster in LV as android container require

Change-Id: If0e83c986ae12b0bb1800b863bf69541ae97bda4
Signed-off-by: Lei wang <leiwan@codeaurora.org>
---
 QcomModulePkg/Library/avb/KeymasterClient.c | 10 ++++++++++
 QcomModulePkg/Library/avb/KeymasterClient.h |  2 ++
 QcomModulePkg/Library/avb/VerifiedBoot.c    |  1 +
 3 files changed, 13 insertions(+)

diff --git a/QcomModulePkg/Library/avb/KeymasterClient.c b/QcomModulePkg/Library/avb/KeymasterClient.c
index 3700ac3..b43c4d1 100644
--- a/QcomModulePkg/Library/avb/KeymasterClient.c
+++ b/QcomModulePkg/Library/avb/KeymasterClient.c
@@ -329,6 +329,16 @@ KeyMasterSetRotAndBootState (KMRotAndBootState *BootState)
 }
 
 EFI_STATUS
+StartKeyMaster (void)
+{
+  /*bring up keymaster*/
+  EFI_STATUS Status = EFI_SUCCESS;
+  KMHandle Handle = {NULL};
+  GUARD (KeyMasterStartApp (&Handle));
+  return Status;
+}
+
+EFI_STATUS
 SetVerifiedBootHash (CONST CHAR8 *Vbh, UINTN VbhSize)
 {
   EFI_STATUS Status = EFI_SUCCESS;
diff --git a/QcomModulePkg/Library/avb/KeymasterClient.h b/QcomModulePkg/Library/avb/KeymasterClient.h
index 6ed1b31..ded7cb4 100644
--- a/QcomModulePkg/Library/avb/KeymasterClient.h
+++ b/QcomModulePkg/Library/avb/KeymasterClient.h
@@ -50,4 +50,6 @@ SetVerifiedBootHash(CONST CHAR8 *Vbh, UINTN VbhSize);
 EFI_STATUS
 KeyMasterGetDateSupport (BOOLEAN *Supported);
 
+EFI_STATUS
+StartKeyMaster (void);
 #endif /* __KEYMASTER_CLIENT_H__ */
diff --git a/QcomModulePkg/Library/avb/VerifiedBoot.c b/QcomModulePkg/Library/avb/VerifiedBoot.c
index fea63bb..2ea5189 100644
--- a/QcomModulePkg/Library/avb/VerifiedBoot.c
+++ b/QcomModulePkg/Library/avb/VerifiedBoot.c
@@ -390,6 +390,7 @@ LoadImageNoAuthWrapper (BootInfo *Info)
 
   GUARD (VBAllocateCmdLine (Info));
   GUARD (LoadImageNoAuth (Info));
+  GUARD (StartKeyMaster ());
 
    if (!IsDynamicPartitionSupport () &&
         !IsRootCmdLineUpdated (Info)) {
-- 
2.7.4

